### ISSUE TITLE
🤔 Is it possible to identify and report if non-reactive value is passed to composable?

### ISSUE THREAD

**Issue by jd-solanki** [vuejs/eslint-plugin-vue#2115]
**Please describe what the rule should do:**
While working with lots of composables that accepts `MaybeRef` type and large project we might accidentally pass the non-reactive value to the composable which likely introduced the bug 🐞

If we can somehow identify this issue and report it, This will be very useful & time saver

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
const val = ref(0)

const { increase } = useCounter(val.value)
```

**Additional context**
N/A

**Comment by ota-meshi** [comment]
The ESLint plugin cannot determine whether a function is composable or not, so it is impossible to fully realize this.
However, there is already a `no-ref-object-destructure` rule for similar purposes.

https://eslint.vuejs.org/rules/no-ref-object-destructure.html
